### ISSUE TITLE
[Suggestion] Eye Emissives

### ISSUE THREAD

**Issue by PlayfulPiano** [gigaherz/EyesInTheDarkness#28]
From what I can tell currently, the eyes are sometimes extremely hard to see, especially if you're using other visual overhauls (e.g. shaders). Making it so the eyes themselves have a bit of an emissive effect could help with pointing them out. Plus it fits with the idea of "glowing eyes in the dark".

**Comment by gigaherz** [comment]
Hi what mc version and mod version are you using? Because I fixed the eyes rendering recently but only for 1.20.1+. I didn't get around to backporting that fix.

**Comment by PlayfulPiano** [comment]
I'm using 1.3.10 for 1.20.1.

**Comment by gigaherz** [comment]
try 1.3.9? 1.3.10 was supposed to make them MORE visible but maybe something went wrong that I didn't consider when applying the fix.

**Comment by PlayfulPiano** [comment]
![image](https://github.com/gigaherz/EyesInTheDarkness/assets/3158887/99eee800-db6f-487a-b6fe-07350b9568e9)
1.3.10

![image](https://github.com/gigaherz/EyesInTheDarkness/assets/3158887/40089272-39a1-4201-9740-7aafb48176f1)
1.3.9

yeah 1.3.10 is def better but it's still like, hard to see dependent on its background.

**Comment by gigaherz** [comment]
Yeah not much I can do about that, then. In 1.3.10 I switched to using the eyes render type, the same one used by invisible spiders. Before that, I already set the eyes to be fully self-lit ("emissive"), so right now they are about as bright as I can make them using vanilla code, so far as I know.

**Comment by PlayfulPiano** [comment]
For example this is what I meant by emissives. Giving the eyes themselves a bit of a glow that's pretty visible.
![image](https://github.com/gigaherz/EyesInTheDarkness/assets/3158887/3880c5d7-722e-410c-904c-58a2a3dceedc)

**Comment by gigaherz** [comment]
> For example this is what I meant by emissives. Giving the eyes themselves a bit of a glow that's pretty visible. ![image](https://user-images.githubusercontent.com/3158887/285278853-3880c5d7-722e-410c-904c-58a2a3dceedc.png)

I assume this is a feature of the shader mod, either hardcoded for specific textures, or using custom metadata for those textures. If it's the latter, you can add that metadata yourself in some resource pack that makes the eyes emissive.
Honestly, I don't particularly care about shader mods, specially optifine, so if you (or someone else) figures out how this works and wants to contribute the necessary metadata then I can look into merging that into the mod, otherwise it's easy to have as a resource pack.

**Comment by PlayfulPiano** [comment]
iirc there are ways for mods to auto-load resource packs built in so it might work out, just would have to mention that emissive effects require some form of mod support (optifine, ETF, iris).

**Comment by gigaherz** [comment]
> iirc there are ways for mods to auto-load resource packs built in so it might work out, just would have to mention that emissive effects require some form of mod support (optifine, ETF, iris).

Mods themselves are resource packs, that's how we add textures and such, we have an assets folder in the .jar that contains resource pack files.

**Comment by PlayfulPiano** [comment]
> > For example this is what I meant by emissives. Giving the eyes themselves a bit of a glow that's pretty visible. ![image](https://user-images.githubusercontent.com/3158887/285278853-3880c5d7-722e-410c-904c-58a2a3dceedc.png)
> 
> I assume this is a feature of the shader mod, either hardcoded for specific textures, or using custom metadata for those textures. If it's the latter, you can add that metadata yourself in some resource pack that makes the eyes emissive. Honestly, I don't particularly care about shader mods, specially optifine, so if you (or someone else) figures out how this works and wants to contribute the necessary metadata then I can look into merging that into the mod, otherwise it's easy to have as a resource pack.

I believe in this case it's from the Entity Model Features mod, which is based on optifine's implementation I believe.

**Comment by PlayfulPiano** [comment]
Entity Texture Features* my bad

**Comment by PlayfulPiano** [comment]
![image](https://github.com/gigaherz/EyesInTheDarkness/assets/3158887/cc728e3f-0f1d-41c5-8e09-eb7006108450)
I'm unsure if under this description the eye texture would qualify or not.

**Comment by PlayfulPiano** [comment]
Oh, actually it does seem like there are emissives.
![image](https://github.com/gigaherz/EyesInTheDarkness/assets/3158887/f60b8975-3f7d-4adf-9f35-599688f720c0)

Wait, do the eyes not render dependent on sky light level? I was mainly noticing my issue while in the nether for example.

**Comment by gigaherz** [comment]
Ah, yes, I forgot, the eyes become transparent when they are lit. More block/sun light = more transparent. At full (level 15) light they are fully invisible.

**Comment by PlayfulPiano** [comment]
Does the transparency affect whether they can attack or not?

It might be worthwhile then to change it so they're not fully transparent within certain dimensions at least, like the nether or end, considering hostiles still spawn at those higher light levels.

**Comment by gigaherz** [comment]
Yes, fully invisible ones don't attack.
And yeah good idea, I should make a dimension filter so that the light level is only used in dimensions with a sun...

**Comment by PlayfulPiano** [comment]
Yeah, that would likely fix my issues with visibility as well.
